src/supertux/player_status.cpp

```diff
@@ -233,32 +257,87 @@ PlayerStatus::read(const ReaderMapping& mapping)
   mapping.get("title-level", title_level);
 }
 
+void
+PlayerStatus::give_item_from_pocket(Player* player)
+{
+  if (!is_item_pocket_allowed())
+    return;
+
+  BonusType bonustype = m_item_pockets[player->get_id()];
+  if (bonustype == NO_BONUS)
+    return;
+
+  m_item_pockets[player->get_id()] = NO_BONUS;
+
+  auto& powerup = Sector::get().add<PocketPowerUp>(bonustype, Vector(0,0));
+  Vector pos = player->get_pos();
+  pos.x = player->get_bbox().get_left();
+  pos.y = player->get_bbox().get_top() - powerup.get_bbox().get_height() - 15;
+  powerup.set_pos(pos);
+}
+
+void
+PlayerStatus::add_item_to_pocket(BonusType bonustype, Player* player)
+{
+  if (!is_item_pocket_allowed())
+    return;
+
+  if (bonustype <= GROWUP_BONUS)
+    return;
+
+  m_item_pockets[player->get_id()] = bonustype;
+}
+
+bool
+PlayerStatus::is_item_pocket_allowed() const
+{
+  if (!GameSession::current())
+  {
+    worldmap::WorldMap* worldmap = worldmap::WorldMap::current();
+    if (worldmap)
+    {
+      return worldmap->is_item_pocket_allowed();
+    }
+    else
+    {
+      // This level is probably in a levelset, pick ON.
+      return true;
+    }
+  }
+
+  Level* level = Level::current();
```

Comment:
Why not `GameSession::current()->get_current_level()`? `Level::current()` is funky IMO. Also you checked if `GameSession::current()` is valid beforehand.

Reply:
Is this value correct when in the editor?

Reply:
Don't think so.